src/probes.cpp

```diff
@@ -404,6 +405,7 @@ void environment_variable_assign(dyntracer_t* dyntracer,
     if ((!state.assignment_stack_is_empty())
     and (state.peek_assignment_stack().get_symbol() == symbol)
```

Comment:
this logic seems incorrect. `get_symbol` is returning a SEXP (pointer) allocated by `strsxp_to_sym`. This will not be the same memory address as that of `symbol` even if their contents are the same. 
The fix is twofold -
1) Do not allocate symbol (remove `get_symbol`)
2) Use string equality rather than pointer equality